### PR TITLE
update login feature

### PR DESCRIPTION
**Pull request by sharefahaneem**

### CODE REVIEW

File: lib/ui/login/widgets/text_field_login_widget.dart [hunk 1]
```diff
@@ -1,22 +1,73 @@
 import 'package:flutter/material.dart';
 
-class TextFieldLoginWidget extends StatelessWidget {
+class TextFieldLoginWidget extends StatefulWidget {
```
Comment (Zer0-01) [inline lib/ui/login/widgets/text_field_login_widget.dart:3]: @sharefahaneem 
why need to change to stateful?

File: lib/ui/login/pages/login_layout.dart [hunk 2]
```diff
@@ -5,54 +5,142 @@ import 'package:flutter_kokohub/ui/login/widgets/login_button_widget.dart';
 import 'package:flutter_kokohub/ui/login/widgets/signup_widget.dart';
 import 'package:flutter_kokohub/ui/login/widgets/text_field_login_widget.dart';
 
-class LoginLayout extends StatelessWidget {
+import '../../../utils/authentication_validator.dart';
+
+class LoginLayout extends StatefulWidget {
```
Comment (Zer0-01) [inline lib/ui/login/pages/login_layout.dart]: @sharefahaneem 
Why need to change to stateful?

File: lib/ui/login/pages/login_layout.dart [hunk 3]
```diff
@@ -5,54 +5,142 @@ import 'package:flutter_kokohub/ui/login/widgets/login_button_widget.dart';
...
-            Padding(
-              padding: EdgeInsets.only(
-                  top: 8.0, bottom: 8.0, left: 24.0, right: 24.0),
-              child: TextFieldLoginWidget(
-                labelText: 'Password',
+      child: SafeArea(
+        child: SingleChildScrollView(
+          child: ConstrainedBox(
+            constraints: BoxConstraints(minHeight: screenHeight),
+            child: IntrinsicHeight(
```
Comment (Zer0-01) [inline lib/ui/login/pages/login_layout.dart:43]: @sharefahaneem 
Can you explain about the ConstrainedBox and Intrinsic Height?

File: lib/ui/login/pages/login_layout.dart [hunk 4]
```diff
@@ -5,54 +5,142 @@ import 'package:flutter_kokohub/ui/login/widgets/login_button_widget.dart';
...
-            SignupWidget(),
-          ],
+          ),
         ),
       ),
     );
   }
 }
+
+class LogoImage extends StatelessWidget {
```
Comment (Zer0-01) [inline lib/ui/login/pages/login_layout.dart]: @sharefahaneem
Is it possible you create this custom widget in another separate dart file?

File: lib/ui/login/pages/login_layout.dart [hunk 5]
```diff
@@ -5,54 +5,142 @@ import 'package:flutter_kokohub/ui/login/widgets/login_button_widget.dart';
...
+      child: Image.asset(
+        "assets/images/login_logo.png",
+        height: MediaQuery.of(context).size.height * 0.3,
+        width: MediaQuery.of(context).size.width * 0.4,
+      ),
+    );
+  }
+}
+
+class EmailFormField extends StatelessWidget {
```
Comment (Zer0-01) [inline lib/ui/login/pages/login_layout.dart]: @sharefahaneem
Is it possible you create this custom widget in another separate dart file?

File: lib/ui/login/pages/login_layout.dart [hunk 6]
```diff
@@ -5,54 +5,142 @@ import 'package:flutter_kokohub/ui/login/widgets/login_button_widget.dart';
...
+        onSaved: (value) {
+          emailController.text = value!;
+        },
+        validator: (value) => AuthenticationValidator.emailValidation(value),
+      ),
+    );
+  }
+}
+
+class PasswordFormField extends StatelessWidget {
```
Comment (Zer0-01) [inline lib/ui/login/pages/login_layout.dart]: @sharefahaneem
Is it possible you create this custom widget in another separate dart file?

File: lib/ui/login/widgets/signup_widget.dart [hunk 7]
```diff
@@ -1,21 +1,26 @@
...
+        style: const TextStyle(color: Colors.black),
         children: <TextSpan>[
           TextSpan(
-            text: 'Sign up',
-            style: TextStyle(
+            text: 'Sign up.',
+            style: const TextStyle(
               color: AppColors.loginButtonColor, // Color of the clickable text
             ),
+            recognizer: TapGestureRecognizer()
```
Comment (Zer0-01) [inline lib/ui/login/widgets/signup_widget.dart:20]: @sharefahaneem
Can you explain about the recognizer?

File: lib/ui/login/widgets/text_field_login_widget.dart (hunk 1 above)
Comment (sharefahaneem) [reply lib/ui/login/widgets/text_field_login_widget.dart:3]: in this widget, we need to use stateful as the email and password will have state changes such as the 'isPassword' property changes the behavior of the input field (whether the text is visible or hidden).
however it possible to not using it as long as we have bloc, but we need further research to change it to bloc

File: lib/ui/login/pages/login_layout.dart (hunk 2 above)
Comment (sharefahaneem) [reply lib/ui/login/pages/login_layout.dart]: same as previous comment which it possible change to bloc. but to clarify, this stateful used for changing the UI based on some state such as show error when key in wrong email or password, show/hide password etc.

extra note: we can prevent using stateful when the state can be manage externally but in rare case we need to use it when the local UI isn't tied to the logic such as controlling animations.

File: lib/ui/login/pages/login_layout.dart (hunk 3 above)
Comment (sharefahaneem) [reply lib/ui/login/pages/login_layout.dart:43]: we need it as to make the layout adapt to various screen sizes
- ConstrainedBox: Ensures the layout fills at least the screen's height (if we dont have this, the container not fill until bottom of the screen)
- IntrinsicHeight: Sizes the column based on its content, preventing unnecessary stretching

File: lib/ui/login/widgets/signup_widget.dart (hunk 7 above)
Comment (sharefahaneem) [reply lib/ui/login/widgets/signup_widget.dart:20]: To make only specific text clickable, such as the 'sign up' text within a longer sentence, we can ensure that only 'sign up' is clickable

File: lib/ui/login/pages/login_layout.dart (hunk 4 above)
Comment (sharefahaneem) [reply lib/ui/login/pages/login_layout.dart]: yup, sure

**Review by Zer0-01 — APPROVED** [review]

**Pull request merged**